src/figuresClasses.ts

```diff
@@ -2,10 +1,76 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side of triangle must be greater than 0');
+    }
+
+    const maxSide = Math.max(a, b, c);
+
+    if (maxSide >= a + b + c - maxSide) {
```

Comment:
The logic for checking the validity of the triangle sides is correct, ensuring the triangle inequality theorem is satisfied.